### PR TITLE
This adds Instance Details view

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #10951 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Review by akus062381 — APPROVED** [review]
looks good!

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        <DetailList gutter="sm">
+          <Detail
+            label={t`Host Name`}
+            value={instance.hostname}
+            dataCy="instance-detail-name"
+          />
+          <Detail
+            label={t`Status`}
+            value={
+              <StatusLabel status={healthCheck?.errors ? 'error' : 'healthy'} />
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js:199]: I think we want to show the health check errors here if an API query returns any. @AlanCoding do you know the format of the `errors` field or have a sample error message we could use to format for display in the UI?

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    isLoading,
+    error: contentError,
+    request: fetchDetails,
+    result: { instance },
+  } = useRequest(
+    useCallback(async () => {
+      const {
+        data: { results },
+      } = await InstanceGroupsAPI.readInstances(instanceGroup.id);
+      let instDetails;
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
      let instanceDetails;
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+      let healthCheckDetails;
+      const isAssociated = results.some(
+        ({ id: instId }) => instId === parseInt(instanceId, 10)
+      );
+
+      if (isAssociated) {
+        const [{ data: instanceDetails }, { data: healthCheckData }] =
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
        const [{ data: details }, { data: healthCheckData }] =
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+          await Promise.all([
+            InstancesAPI.readDetail(instanceId),
+            InstancesAPI.readHealthCheckDetail(instanceId),
+          ]);
+
+        instDetails = instanceDetails;
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
        instanceDetails = details;
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        healthCheckDetails = healthCheckData;
+      } else {
+        throw new Error(
+          `This instance is not associated with this instance group`
+        );
+      }
+
+      setBreadcrumb(instanceGroup, instDetails);
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
      setBreadcrumb(instanceGroup, instanceDetails);
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+      setHealthCheck(healthCheckDetails);
+      setForks(
+        computeForks(
+          instDetails.mem_capacity,
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
          instanceDetails.mem_capacity,
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+          instDetails.cpu_capacity,
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
          instanceDetails.cpu_capacity,
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+          instDetails.capacity_adjustment
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
          instanceDetails.capacity_adjustment
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+        )
+      );
+      return { instance: instDetails };
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
      return { instance: instanceDetails };
```

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+            />
+          )}
+        </DetailList>
+        <CardActionsRow>
+          <Tooltip content={t`Run a health check on the instance`}>
+            <Button
+              isDisabled={!me.is_superuser}
+              variant="primary"
+              ouiaId="health-check-button"
+              onClick={() => fetchHealthCheck()}
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js]:
```suggestion
              onClick={fetchHealthCheck}
```
For SyntheticEvents you can pass the function name. I would only use arrow functions if you need to pass in parameters.

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.test.js [hunk 11]
```diff
@@ -0,0 +1,389 @@
...
+      );
+    });
+    await waitForElement(wrapper, 'ContentLoading', (el) => el.length === 0);
+    expect(wrapper.find('ContentError')).toHaveLength(1);
+    expect(InstanceGroupsAPI.readInstances).toBeCalledWith(2);
+    expect(InstancesAPI.readHealthCheckDetail).not.toBeCalled();
+    expect(InstancesAPI.readDetail).not.toBeCalled();
+  });
+
+  test('Should call disassociate', async () => {
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.test.js:401]: Can we also add a test for the Health Check button here?

**Review by kialam — CHANGES_REQUESTED** [review]
This looks really good so far! Aside from some variable naming nitpicks, my only two requests are to handle health check error message rendering in the UI and adding a unit test for the new Health Check button.

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js (hunk 1 above)
Comment (AlanCoding) [reply awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js:199]: example

```
Receptor error getting worker info from receptor-2, detail:
exec: "ansible-runner": executable file not found in $PATH
```

From API

```
"errors": "Receptor error getting worker info from receptor-2, detail:\nexec: \"ansible-runner\": executable file not found in $PATH",
```

Comment (AlexSCorey) [reply awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js:199]: @kialam the Health Check errors are rendered [here](https://github.com/ansible/awx/pull/11069/files#diff-bd47c736dddf4c8b375895668ba9e3a8143af17a8aaf32c48536a3a25d322de1R253).  This is just rendering the status of the instance.

Comment (kialam) [reply awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.js:199]: Ah excellent. Confirmed errors look good in the UI:
Screen Shot 2021-09-15 at 11 05 25 AM

File: awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.test.js [hunk 12]
```diff
@@ -0,0 +1,490 @@
...
+      wrapper.find("Button[ouiaId='health-check-button']").prop('onClick')();
+    });
+    expect(InstancesAPI.createHealthCheck).toBeCalledWith(1);
+    wrapper.update();
+    expect(
+      wrapper.find("Detail[label='Last Health Check']").prop('value')
+    ).toBe('9/15/2021, 6:02:07 PM');
+  });
+
+  test('Should handle api error for health check', async () => {
```
Comment (kialam) [inline awx/ui/src/screens/InstanceGroup/InstanceDetails/InstanceDetails.test.js:352]: nice

**Review by kialam — APPROVED** [review]
LGTM. Great job Alex!

**Pull request merged**